docs/knuu-new.md

```diff
@@ -0,0 +1,161 @@
+- `Logger`: A logger instance for logging (_default: logrus with info level_).
+
+### Example
+
+```go
+package main
+
+import (
+    "context"
+    "log"
+    "time"
+
+    "github.com/celestiaorg/knuu/pkg/k8s"
+    "github.com/celestiaorg/knuu/pkg/minio"
+    "github.com/celestiaorg/knuu/pkg/builder"
+    "github.com/sirupsen/logrus"
+    "github.com/celestiaorg/knuu"
+)
+
+func main() {
+    ctx := context.Background()
+
+    options := knuu.Options{
+        Scope:        "example-scope",
+        Timeout:      30 * time.Minute,
+        Logger:       logrus.New().WithLevel(logrus.ErrorLevel),
+    }
+
+    kn, err := knuu.New(ctx, options)
+    if err != nil {
+        log.Fatalf("Failed to create Knuu object: %v", err)
+    }
+    kn.HandleStopSignal(ctx) // explained in the next section
+
+    defer func() {
+        err := kn.CleanUp(ctx)
+        if err != nil {
+            log.Fatalf("Failed to clean up resources: %v", err)
+        }
+    }()
+
+
+    sampleInstance, err := kn.NewInstance("alpine")
+    if err != nil {
+        log.Fatalf("Failed to create instance: %v", err)
+    }
+
+    err = sampleInstance.Build().SetGitRepo(ctx, builder.GitContext{
+		Repo:     "https://github.com/<sample-repo>.git",
+		Branch:   "<desired-branch>",
+		Username: "<git-username>", // Leave empty if repo is public
+		Password: "<git-password>", // Leave empty if repo is public
+	})
+    if err != nil {
+        log.Fatalf("Failed to build from a git repo: %v", err)
+    }
+
+    // optionally can set image directly instead of building from git repo
+    // err = sampleInstance.Build().SetImage(ctx, "docker.io/alpine:latest")
+    // if err != nil {
+    //     log.Fatalf("Failed to set image: %v", err)
+    // }
+
+    err = sampleInstance.Build().SetStartCommand("<start-command>", "<arg1>", "<arg2>",...)
+    if err != nil {
+        log.Fatalf("Failed to set start command: %v", err)
+    }
+
+    err = sampleInstance.Build().SetEnvironmentVariable("<env-var-name>", "<env-var-value>")
+    if err != nil {
+        log.Fatalf("Failed to set env var: %v", err)
+    }
+
+    // Adding file before commit will add it to the builder
+    // Therefore the image will be rebuilt with the new file
+    err = sampleInstance.Storage().AddFile("<source-path>", "<destination-path>", "<permissions>")
+    if err != nil {
+        log.Fatalf("Failed to add file: %v", err)
+    }
+
+    err = sampleInstance.Build().Commit(ctx)
+    if err != nil {
+        log.Fatalf("Failed to commit: %v", err)
+    }
+
+    // Adding file after commit will add it to the deployment of the instance
+    // Therefore the is no image rebuilt, however the file must be very small (config maps are used)
+    // and should not be used to transport large files otherwise the deployment will fail
+    err = sampleInstance.Storage().AddFile("<source-path>", "<destination-path>", "<permissions>")
```

Comment:
How should large files be handled? Do we have more info on what classifies larger? 1MB, 10MB, 100MB?

Reply:
Well the max size for a k8s API request is 3MB and config maps are arapped in an API call request, therefore it is in the magnitude of KBs. and it is mainly used for config files like a json or toml ...